v-next/hardhat-utils/src/subprocess.ts

```diff
@@ -0,0 +1,32 @@
+import { spawn } from "node:child_process";
+
+/**
+ * Spawns a detached subprocess to execute a given file with optional arguments.
+ *
+ * @param absolutePathToSubProcessFile - The absolute path to the subprocess file to be executed.
+ * @param args - Optional list of string arguments to pass to the subprocess.
+ * @param env - Optional environment key-value pairs to pass to the subprocess.
+ *
+ * The subprocess runs in a detached mode and its standard input/output is ignored.
+ * This function does not wait for the subprocess to complete and the subprocess is unreferenced
+ * to allow the parent process to exit independently.
+ */
+export function spawnDetachedSubProcess(
+  absolutePathToSubProcessFile: string,
+  args: string[] = [],
+  env: Record<string, string> = {},
+): void {
+  const subprocessArgs = [absolutePathToSubProcessFile, ...args];
+
+  if (absolutePathToSubProcessFile.endsWith(".ts")) {
+    subprocessArgs.unshift("--import", "tsx/esm");
+  }
+
+  const subprocess = spawn(process.execPath, subprocessArgs, {
+    detached: true,
+    env,
+    stdio: "ignore",
```

Comment:
At least one of the tests should print to stdout and stderr to validate that this works

Reply:
Done in [this commit] (https://github.com/NomicFoundation/hardhat/pull/5471/commits/b3ed594afe2438f7585cb8834b7359a829e973f4) with other improvements and[ this commit](https://github.com/NomicFoundation/hardhat/pull/5471/commits/b7dd913a82ab58d71a5f603fee8a661e14a1bfac)

Reply:
@alcuadrado windows tests passed with console.log and console.error in the subprocess. I guess that this means that everything is ok (?). 
Link to subprocess code: https://github.com/NomicFoundation/hardhat/pull/5471/files#r1665894434

Reply:
great!